src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,541 @@
+            if i === nothing
+                push!(fanRays,coneRay)
+                push!(coneIncidence,length(fanRays))
+            else
+                push!(coneIncidence,i)
+            end
+        end
+        push!(fanIncidences,coneIncidence)
+    end
+
+    # convert Vector{Vector} to IncidenceMatrix and Matrix respectively
+    fanIncidences = IncidenceMatrix(fanIncidences)
+    fanRays = matrix(QQ,fanRays)
+
+    fanLineality = matrix(QQ,lineality_space(first(listOfCones)))
+
+    return PolyhedralFan(fanRays,fanLineality,fanIncidences)
+end
+
+
+
+
+@doc raw"""
+    groebner_fan(I::MPolyIdeal; return_groebner_bases::Bool=false, return_orderings::Bool=false, verbose_level::Int=0)
+
+Return a `PolyhedralFan` representing the Groebner fan of `I`, where is a multivariate polynomial ideal over a field.
+
+If `verbose_level` is positive, also print how many cones have been computed during the traversal.
+
+If `I` is not weighted homogeneous with respect to a positive weight vector, the Groebner fan will be restricted to the positive orthant.  Otherwise, the Groebner fan will span the entire space and have a non-trivial lineality space.
+
+If `return_groebner_bases==true`, also return a dictionary whose keys are interior points of the maximal cones and whose values are the Groebner bases for those cones.  Their union will be a universal Groebner basis.
+
+If `return_orderings==true`, returns a dictionary whose keys are interior points of the maximal Groebner cones and whose values are monomial orderings for those cones.  These orderings are suboptimal and hence it is generally recommended to create new orderings with the interior points.  However they do contain information on how the fan was traversed.
+
+# Examples
+```jldoctest
+julia> Qx,(x1,x2,x3) = PolynomialRing(QQ,["x1","x2","x3"])
+(Multivariate Polynomial Ring in x1, x2, x3 over Rational Field, QQMPolyRingElem[x1, x2, x3])
+
+julia> I = ideal([x1,x2+x3])
+ideal(x1, x2 + x3)
+
+julia> SigmaI = groebner_fan(I)
+Polyhedral fan in ambient dimension 3
+
+julia> SigmaI,gbs,ords = groebner_fan(I,return_groebner_bases=true,return_orderings=true)
+3-element Vector{Any}:
+ Polyhedral fan in ambient dimension 3
+ Dict{Vector{ZZRingElem}, Vector{QQMPolyRingElem}}([0, -1, 0] => [x1, x2 + x3], [0, 0, -1] => [x2 + x3, x1])
+ Dict{Vector{ZZRingElem}, MonomialOrdering{QQMPolyRing}}([0, -1, 0] => matrix_ordering([x1, x2, x3], [1 1 1])*matrix_ordering([x1, x2, x3], [0 0 0])*matrix_ordering([x1, x2, x3], [0 -1 1])*revlex([x1, x2, x3]), [0, 0, -1] => degrevlex([x1, x2, x3]))
+
+```
+"""
+
+function groebner_fan(I::MPolyIdeal; return_groebner_bases::Bool=false, return_orderings::Bool=false, verbose_level::Int=0)
+
+    ###
+    # Preparation:
+    #   Test whether the ideal is weighted homogeneous with respect to a positive weight vector
+    #   Preference is given to the ones vector for the sake of optimisation
+    ###
+    G = groebner_basis(I,complete_reduction=true)
+    ord = G.ord
```

Comment:
Presumably here you want to use your new `ordering` method?
```suggestion
    ord = ordering(G)
```